internal/api/server.go

```diff
@@ -2,107 +2,278 @@ package api
-		ManagementClusterId: managementClusterId,
-		Private:             false,
+
+	probe := &v1.ProbeObject{}
+	err = json.Unmarshal([]byte(cm.Data["probe-config.json"]), probe)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal probe from configmap: %w", err)
+	}
+	return probe, nil
+}
+
+func (k *KubernetesProbeStore) CreateProbe(ctx context.Context, probe v1.ProbeObject, urlHashString string) (*v1.ProbeObject, error) {
+	payloadBytes, err := json.Marshal(probe)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal payload: %w", err)
+	}
+
+	configMapName := fmt.Sprintf(probeConfigMapNameFormat, probe.Id)
+	cmLabels := make(map[string]string)
+	if probe.Labels != nil {
+		for key, val := range *probe.Labels {
+			cmLabels[key] = val
+		}
 	}
+	// Add our base app label from the constant
+	cmLabels[baseAppLabelKey] = baseAppLabelValue
+	cmLabels[probeURLHashLabelKey] = urlHashString
 
-	responseStruct := v1.ProbesArrayResponse{
-		Probes: []v1.ProbeObject{
-			dummyProbe,
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      configMapName,
+			Namespace: k.Namespace,
+			Labels:    cmLabels,
 		},
+		Data: map[string]string{
+			"probe-config.json": string(payloadBytes),
+		},
+	}
+
+	_, err = k.Client.CoreV1().ConfigMaps(k.Namespace).Create(ctx, configMap, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
 	}
+	return &probe, nil
+}
 
-	return v1.ListProbes200JSONResponse(responseStruct), nil
+func (k *KubernetesProbeStore) DeleteProbe(ctx context.Context, probeID uuid.UUID) error {
+	configMapName := fmt.Sprintf(probeConfigMapNameFormat, probeID)
+	return k.Client.CoreV1().ConfigMaps(k.Namespace).Delete(ctx, configMapName, metav1.DeleteOptions{})
 }
 
-// (GET /metrics/probe/{cluster_id})
-func (Server) GetProbeById(ctx context.Context, request v1.GetProbeByIdRequestObject) (v1.GetProbeByIdResponseObject, error) {
-	// Fake respone while we wire things together
-	managementClusterId, err := uuid.Parse("957c5277-f74c-4b24-938a-f70bab28aab5")
+func (k *KubernetesProbeStore) ProbeWithURLHashExists(ctx context.Context, urlHashString string) (bool, error) {
+	hashLabelSelector := fmt.Sprintf("%s=%s", probeURLHashLabelKey, urlHashString)
+	existingProbes, err := k.Client.CoreV1().ConfigMaps(k.Namespace).List(ctx, metav1.ListOptions{
+		LabelSelector: hashLabelSelector,
+	})
 	if err != nil {
-		return v1.GetProbeById404JSONResponse{}, nil
+		return false, fmt.Errorf("failed to check for existing probes: %w", err)
+	}
+	return len(existingProbes.Items) > 0, nil
+}
+
+// Server is the main API server object.
+type Server struct {
+	Store ProbeStorage
+}
+
+// NewServer creates a new API server.
+func NewServer(store ProbeStorage) Server {
+	return Server{
+		Store: store,
 	}
-	dummyProbe := v1.ProbeObject{
-		Id:                  request.ClusterId,
-		ApiserverUrl:        "https://v1.example.com/cluster-1",
-		ManagementClusterId: managementClusterId,
-		Private:             false,
+}
```

Comment:
Do you think it's worth separating this out to it's own package?

Reply:
I suppose if we make a `probestore` package that will make it easier to support a different "database" backend in the future if we ever needed (something like postgres). 

I can work on that refactor if you agree